notebooks/plotly_roc_plot.py

```diff
@@ -4,43 +4,129 @@
 if get_ipython() is not None:
     get_ipython().magic('load_ext autoreload')
     get_ipython().magic('autoreload 2')
+    if "arthur" not in __file__:
+        __file__ = '/Users/adria/Documents/2023/ACDC/Automatic-Circuit-Discovery/notebooks/plotly_roc_plot.py'
 
 import plotly
 import os
+import numpy as np
 import json
 import wandb
+import time
 from plotly.subplots import make_subplots
 import plotly.graph_objects as go
 from pathlib import Path
+import plotly.express as px
+import pandas as pd
+import argparse
 
-JSON_FNAME = Path(__file__).parent.parent / "acdc" / "media" / "roc_data.json"
+from notebooks.emacs_plotly_render import set_plotly_renderer
+set_plotly_renderer("emacs")
 
 # %%
 
-tasks = ["Circuit Recovery (IOI)", "Tracr (Reverse)", "Tracr (Proportion)", "Docstring", "Greater Than"]
-
-fig = make_subplots(
-    rows=2,
-    cols=4,
-    # specs parameter is really cool, this argument needs to have same dimenions as the rows and cols
-    specs=[[{"rowspan": 2, "colspan": 2}, None, {}, {}], [None, None, {}, {}]],
-    print_grid=True,
-    # subplot_titles=("First Subplot", "Second Subplot", "Third Subplot", "Fourth Subplot", "Fifth Subplot"),
-    subplot_titles=tuple(tasks),
-    x_title="False positive rate (edges)",
-    y_title="True positive rate (edges)",
-    # title_font=dict(size=8),
-)
-
-fig.update_annotations(font_size=12)
-
-rows_and_cols = [
-    (1, 1),
-    (1, 3), 
-    (1, 4),
-    (2, 3),
-    (2, 4),
-]
+parser = argparse.ArgumentParser()
+parser.add_argument('--arrows', action='store_true', help='Include help arrows')
+
+if get_ipython() is not None:
+    args = parser.parse_args([])
+else:
+    args = parser.parse_args()
+
+# %%
+
+def pessimistic_auc(xs, ys):
+    i = np.argsort(xs)
```

Comment:
```suggestion
    # Sort indices based on 'x' and 'y'
    i = np.lexsort((ys, xs)) # lexsort sorts by the last column first, then the second last, etc., i.e we firstly sort by x and then y to break ties

```

without this, the function is broken

Reply:
Ah I guess you're right and this could make the asserts fail sometimes while dthe data is indeed monotonic.